File: Implement_Queue_by_Two_Stacks.cpp
class Queue {
public:
    stack<int> stack1;
    stack<int> stack2;

    Queue() {
        // do intialization if necessary
    }

    void push(int element) {
        // write your code here
        stack1.emplace(element);
    }
    
    int pop() {
        // write your code here
        int ret = top();
        stack2.pop();
        return ret;
    }

    int top() {
        // write your code here
        if (stack2.empty())
        {
            while (!stack1.empty())
            {
                stack2.emplace(stack1.top());
                stack1.pop();
            }
        }
        return stack2.top();
    }
};

File: Implement Queue by Two Stacks.py
class MyQueue:

    def __init__(self):
        self.stack1 = [] #in
        self.stack2 = [] #out
        
    def push(self, element):
        # write your code here
        self.stack1.append(element)
        

    def top(self):
        # write your code here
        # return the top element
        self.__touch()
        return self.stack2[-1]

    def pop(self):
        # write your code here
        # pop and return the top element
        self.__touch()
        return self.stack2.pop()
        
    def __touch(self):
        if self.stack2:
            return
        while self.stack1:
            self.stack2.append(self.stack1.pop())